configs/typescript-react/package.json

```diff
@@ -27,9 +27,13 @@
 		"typescript": "^5.6.2"
 	},
 	"peerDependencies": {
+		"react": "^18.3.1 || ^19.0.0",
+		"react-dom": "^18.3.1 || ^19.0.0",
 		"typescript": "^5.6.2"
 	},
 	"dependencies": {
-		"@canonical/typescript-config-base": "^0.1.1-experimental.0"
+		"@types/react": "^18.3.12 || npm:types-react-dom@rc",
+		"@types/react-dom": "^18.3.1 || npm:types-react-dom@rc",
```

Comment:
These were added explicitly.

Previously, they were being indirectly installed by one of our dependencies.

If these dependencies are not present, errors like this occur:

```
      src/ui/Page/Page.tsx(1,19): error TS7016: Could not find a declaration file for module 'react'. '/Users/jmuzina/source/work/canonical/repos/ds25/node_modules/react/index.js' implicitly has an 'any' type.
        Try `npm i --save-dev @types/react` if it exists or add a new declaration (.d.ts) file containing `declare module 'react';`
        Try `npm i --save-dev @types/react` if it exists or add a new declaration (.d.ts) file containing `declare module 'react/jsx-runtime';`
      src/ui/Page/Page.tsx(81,15): error TS7026: JSX element implicitly has type 'any' because no interface 'JSX.IntrinsicElements' exists.
```